src/resmom/linux/alps.c

```diff
@@ -3727,21 +3727,22 @@ inventory_loop_on_segments(basil_node_t *node, vnl_t *nv, char *arch,
 						goto bad_vnl;
 					if (accel->data.gpu->memory) {
 						attr = "resources_available.accelerator_memory";
-						if (totseg == 0) {
+						if (totseg == 0 || vnode_per_numa_node) {
 							snprintf(utilBuffer,
 								sizeof(utilBuffer),
 								"%umb",
 								accel->data.gpu->memory);
-						} else if (vnode_per_numa_node) {
-							snprintf(utilBuffer,
-								sizeof(utilBuffer),
-								"@%s_%ld_0",
-								mpphost, node->node_id);
+							if (vnode_per_numa_node && !totseg) {
+								snprintf(utilBuffer,
+									 sizeof(utilBuffer),
+									 "@%s_%ld_0",
+									 mpphost, node->node_id);
+							}
+							if (vn_addvnr(nv, vname, attr,
+								      utilBuffer, 0,
+								      0, NULL) == -1)
+								goto bad_vnl;
```

Comment:
I don't understand why you changed this code from what had previously existed.  Looks like now with vnode_per_numa_node set to true, the first NUMA node vnode will get accelerator memory set to @host_nodeid_0 (i.e. itself).  And the value for resources_available.accelerator_memory would be set on all other NUMA node vnodes.  I couldn't tell if pbs_cray_vnode_per_numa PTL had passed with this change...the latest results are older than this modification.  Did the pbs_cray_vnode_per_numa.py PTL test pass with this change?  If so, it seems like the test may need some modifying.  :)